CleanGenerator/Templates/TestEndpoints/TestEndpointsTemplate.tt

```diff
@@ -17,45 +19,67 @@ internal static class <#= _model.EntityTypeName #>HttpClientExtensions
+internal class Create<#= _model.EntityTypeName #>Endpoint : ApiEndpointBase<Create<#= _model.EntityTypeName #>Dto, int>
 {
     internal Create<#= _model.EntityTypeName #>Endpoint(HttpClient httpClient) :
         base(httpClient)
     {
     }
 
-    internal override string BuildPath(Create<#= _model.EntityTypeName #>Dto dto) => $"api/<#= _model.ApiBasePath #>";
+    public override async Task<HttpResponseMessage> Call(Create<#= _model.EntityTypeName #>Dto dto)
+    {
+        var path = $"api/<#= _model.ApiBasePath #>";
+
+        return await HttpClient.PostAsJsonAsync(path, dto);
+    }
 }
 
-internal class Get<#= _model.EntityTypeName #>ByIdEndpoint : GetApiEndpointWithDto<int, <#= _model.EntityTypeName #>Dto>
+internal class Get<#= _model.EntityTypeName #>ByIdEndpoint : ApiEndpointBase<int, <#= _model.EntityTypeName #>Dto>
 {
     internal Get<#= _model.EntityTypeName #>ByIdEndpoint(HttpClient httpClient) :
         base(httpClient)
     {
     }
 
-    internal override string BuildPath(int <#= _model.EntityVariableName #>Id) => $"api/<#= _model.ApiBasePath #>/{<#= _model.EntityVariableName #>Id}";
+    public override async Task<HttpResponseMessage> Call(int <#= _model.EntityVariableName #>Id)
+    {
+        var path = $"api/<#= _model.ApiBasePath #>/{<#= _model.EntityVariableName #>Id}";
+
+        return await HttpClient.GetAsync(path);
+    }
 }
 
-internal class List<#= _model.EntityTypeName #>sEndpoint : GetApiEndpoint<List<<#= _model.EntityTypeName #>Dto>>
+internal class List<#= _model.EntityTypeName #>sEndpoint : ApiEndpointBase<PaginatedListQuery, PaginatedListResponse<<#= _model.EntityTypeName #>Dto>>
 {
     internal List<#= _model.EntityTypeName #>sEndpoint(HttpClient httpClient) :
-        base(httpClient, "api/<#= _model.ApiBasePath #>")
+        base(httpClient)
     {
     }
+
+    public override async Task<HttpResponseMessage> Call(PaginatedListQuery dto)
+    {
+        var path = $"api/<#= _model.EntityVariableName #>?{dto.ToQueryString()}";
+
+        return await HttpClient.GetAsync(path);
+    }
 }
 
-internal class Update<#= _model.EntityTypeName #>Endpoint : PutApiEndpointWithDto<Update<#= _model.EntityTypeName #>Dto, int>
+internal class Update<#= _model.EntityTypeName #>Endpoint : ApiEndpointBase<Update<#= _model.EntityTypeName #>Dto, int>
 {
     internal Update<#= _model.EntityTypeName #>Endpoint(HttpClient httpClient) :
         base(httpClient)
     {
     }
 
-    internal override string BuildPath(Update<#= _model.EntityTypeName #>Dto dto) => $"api/<#= _model.ApiBasePath #>/{dto.Id}";
+    public override async Task<HttpResponseMessage> Call(Update<#= _model.EntityTypeName #>Dto dto)
+    {
+        var path = $"api/<#= _model.ApiBasePath #>/{dto.Id}";
+
+        return await HttpClient.PutAsJsonAsync(path, dto);
+    }
 }
 
-internal class Delete<#= _model.EntityTypeName #>Endpoint : DeleteApiEndpoint
+internal class Delete<#= _model.EntityTypeName #>Endpoint : ApiEndpointBase<int, int>
```

Comment:
I think this should inherit ApiEndpointBaseWithoutResponse as Delete endpoint doesn't return anything.